examples/ssd/train.py

```diff
@@ -0,0 +1,179 @@
+            {'loss': loss, 'loss/loc': loc_loss, 'loss/conf': conf_loss},
+            self)
+
+        return loss
+
+
+class Transform(object):
+
+    def __init__(self, coder, size, mean):
+        # to send cpu, make a copy
+        self.coder = copy.copy(coder)
+        self.coder.to_cpu()
+
+        self.size = size
+        self.mean = mean
+
+    def __call__(self, in_data):
+        # There are five data augmentation steps
+        # 1. Color augmentation
+        # 2. Random expansion
+        # 3. Random cropping
+        # 4. Resizing with random interpolation
+        # 5. Random horizontal flipping
+
+        img, bbox, label = in_data
+
+        # 1. Color augmentation
+        img = random_distort(img)
+
+        # 2. Random expansion
+        if np.random.randint(2):
+            img, param = transforms.random_expand(
+                img, fill=self.mean, return_param=True)
+            bbox = transforms.translate_bbox(
+                bbox, y_offset=param['y_offset'], x_offset=param['x_offset'])
+
+        # 3. Random cropping
+        img, param = random_crop_with_bbox_constraints(
+            img, bbox, return_param=True)
+        bbox, param = transforms.crop_bbox(
+            bbox, y_slice=param['y_slice'], x_slice=param['x_slice'],
+            allow_outside_center=False, return_param=True)
+        label = label[param['index']]
+
+        # 4. Resizing with random interpolatation
+        _, H, W = img.shape
+        img = resize_with_random_interpolation(img, (self.size, self.size))
+        bbox = transforms.resize_bbox(bbox, (H, W), (self.size, self.size))
+
+        # 5. Random horizontal flipping
+        img, params = transforms.random_flip(
+            img, x_random=True, return_param=True)
+        bbox = transforms.flip_bbox(
+            bbox, (self.size, self.size), x_flip=params['x_flip'])
+
+        # Preparation for SSD network
+        img -= self.mean
+        mb_loc, mb_label = self.coder.encode(bbox, label)
+
+        return img, mb_loc, mb_label
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--batchsize', type=int, default=32)
+    parser.add_argument('--gpu', type=int, default=-1)
+    parser.add_argument('--out', default='result')
+    parser.add_argument('--resume')
+    args = parser.parse_args()
+
+    model = SSD300(
+        n_fg_class=len(voc_detection_label_names),
+        pretrained_model='imagenet')
+    model.use_preset('evaluate')
+    train_chain = MultiboxTrainChain(model)
+    if args.gpu >= 0:
+        chainer.cuda.get_device(args.gpu).use()
+        model.to_gpu()
+
+    train = TransformDataset(
+        ConcatenatedDataset(
+            VOCDetectionDataset(year='2007', split='trainval'),
+            VOCDetectionDataset(year='2012', split='trainval')
+        ),
+        Transform(model.coder, model.insize, model.mean))
+    train_iter = chainer.iterators.MultiprocessIterator(
+        train, args.batchsize, n_processes=2)
```

Comment:
Yes, I think `None` is better. `2` was for my testing environment. I should not put it in public code.